docs/api/filters.rst

```diff
@@ -10,4 +13,104 @@ Filters
+
+To ease combining filters together, inspired by the ``transforms`` module of torchvision [11]_, we implemented the ``Compose`` class in the ``filters`` subpackage. ``Compose`` allows ``histolab`` filters - both image and morphological filters - to be concatenated as a chain of functions, without any intermediate transformation or prior knowledge on their ordering. In order to enable this composition, filters have been designed to follow the same usage pattern, and to input/output either a PIL Image or a NumPy array object.
+To clarify how the ``Compose`` object works, let us consider the sequential application of a set of filters sequentially, where the output of a filter is the input of the following one:
+
+.. code-block:: python3
+
+        from PIL import Image
+        from histolab.filters.image_filters import RgbToGrayscale, OtsuThreshold, ApplyMaskImage
+        from histolab.filters.morphological_filters import BinaryDilation
+        def not_composed_filters(image_rgb):
+            rgb_to_grayscale = RgbToGrayscale()
+            otsu_threshold = OtsuThreshold()
+            binary_dilation = BinaryDilation()
+            apply_mask_image = ApplyMaskImage(image_rgb) # apply the resulting mask on the original image
+
+            image_gray = rgb_to_grayscale(image_rgb)
+            image_thresholded = otsu_threshold(image_gray)
+            image_dilated = binary_dilation(image_thresholded)
+            return apply_mask_image(image_dilated)
+        image_rgb = Image.open("path/to/image.png")
+        resulting_image = not_composed_filters(image_rgb)
+
+
+Despite being formally correct, the above implementation in neither intuitive or economic (in terms of memory used and lines of code). The use of a ``Compose`` object leads to a more compact implementation:
+
+.. code-block:: python3
+
+    from PIL import Image
+    from histolab.filters.image_filters import RgbToGrayscale, OtsuThreshold, ApplyMaskImage, Compose
+    from histolab.filters.morphological_filters import BinaryDilation
+
+    def composed_filters(image_rgb):
+        filters = Compose([
+            RgbToGrayscale(),
+            OtsuThreshold(),
+            BinaryDilation(),
+            ApplyMaskImage(image_rgb),
+        ])
+        return filters(image_rgb)
+
+    image_rgb = Image.open("path/to/image.png")
+    resulting_image = composed_filters(image_rgb)
+
+.. note::
+    Although ``not_composed_filters`` and ``composed_filters`` functions return the same result, the use of ``Compose`` avoids storing intermediate results and wasting memory in case of very large input image. For example, the peak RAM usage for an image of size 19,394px x 6,136px (84.9 MB) is ~2600 MB when using ``composed_filters`` in contrast to the ~3200 MB allocated for the ``not_composed_filters`` function.
+
+Largest Tissue Box extraction
```

Comment:
I would move this somewhere in the tiler module

Reply:
tiler?

Reply:
oh the biggest box you mean

Reply:
yep